### PR TITLE
[doc] How-to: Add a confirm dialog in a form

### PR DESCRIPTION
**Pull request by dpobel**
Documentation for https://jira.ez.no/browse/EZP-27573

### CODE REVIEW

File: docs/howto/add_confirm_dialog_in_a_form.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# How-to: Add a confirm dialog in a form
+
+Asking the user for a confirmation before doing a *dangerous* operation is a
```
Comment (bdunogier) [inline docs/howto/add_confirm_dialog_in_a_form.md]: "dangerous" sounds a bit dreadful in that context.

File: docs/howto/add_confirm_dialog_in_a_form.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+    <button type="submit" name="DangerousOperation">
+        Dangerous operation
+    </button>
+</form>
+```
+
+To show a confirm dialog before doing the *Dangerous operation*, you have to:
+
+1. Replace the *DangerousOperation* `submit` button by a button with:
```
Comment (bdunogier) [inline docs/howto/add_confirm_dialog_in_a_form.md]: I think it's "with", not "by".

And if you make that "with a button having the following attributes:", you can remove "the ... attribute" from the bullet points below, and avoid repeating it.

File: docs/howto/add_confirm_dialog_in_a_form.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      submit the form
+    * the `class` attribute containing `ez-js-open-modal` so the app recognizes
+      this button as a button to open a dialog.
+    * the `value` attribute set with a selector matching the `<dialog>` element
+      to show
+1. Add the actual `<dialog>` element in the form. It should contain the
+   original submit button with the class `ez-js-close-modal` so that clicking on
+   it submits the form and closes the dialog.
+
+This gives the following HTML code:
```
Comment (bdunogier) [inline docs/howto/add_confirm_dialog_in_a_form.md]: I think that "gives" in that context is a bit of a frenchism :)

File: docs/howto/add_confirm_dialog_in_a_form.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+```twig
+{{ form_start(myForm, {'action': path('form_handling_route')}) }}
+    <!-- some markup -->
+
+    {{ form_widget(myForm.delete, {'label': 'Dangerous Operation'}) }}
+{{ form_end(myForm) }}
+```
+
+The changes explained in the previous section can be applied directly in the
+Twig template but to ease the addition of a confirm dialog in such context, the
```
Comment (bdunogier) [inline docs/howto/add_confirm_dialog_in_a_form.md]: I'd end the sentence after "Twig template", then start a new one:

"[...] can be applied directly in the Twig template. To ease the addition of a confirm dialog in such context, the"

("to ease the addition" sounds a bit "heavy" to me. You could maybe just remove it:

"[...] "can be applied directly in the Twig template. The Hybrid UI application provides [...]"

But @ezsystems/documentation-team probably has better insight than me on this :)

**Review by bdunogier — APPROVED** [review]
Couple grammar suggestions, but no blockers.

**Comment by bdunogier** [comment]
What would you think of adding a couple links to places in the code where this is implemented ? I know that those places might change, but it still sounds very profitable for developers. It also encourages them to check the source code out.

File: docs/howto/add_confirm_dialog_in_a_form.md (hunk 3 above)
Comment (DominikaK) [reply docs/howto/add_confirm_dialog_in_a_form.md]: "results in" would be 13% less French ;)

File: docs/howto/add_confirm_dialog_in_a_form.md (hunk 4 above)
Comment (DominikaK) [reply docs/howto/add_confirm_dialog_in_a_form.md]: I'd go with: 
"The changes explained in the previous section can be applied directly in the Twig template. The Hybrid Platform UI application provides the template `components/confirm_delete_dialog.html.twig` to make adding a confirm dialog in such context easier."

File: docs/howto/add_confirm_dialog_in_a_form.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        </footer>
+    </dialog>
+</form>
+```
+
+### Symfony form and Twig template
+
+In a Symfony application, it's common to use [the Form Component to integrate
+forms](https://symfony.com/doc/current/forms.html) and most forms in the
+application are build this way. The Hybrid Platform UI application is also
```
Comment (DominikaK) [inline docs/howto/add_confirm_dialog_in_a_form.md]: "are buil**t** this way"

File: docs/howto/add_confirm_dialog_in_a_form.md (hunk 4 above)
Comment (bdunogier) [reply docs/howto/add_confirm_dialog_in_a_form.md]: See, told you :)

File: docs/howto/add_confirm_dialog_in_a_form.md (hunk 1 above)
Comment (dpobel) [reply docs/howto/add_confirm_dialog_in_a_form.md]: @bdunogier any suggestion ?

Comment (bdunogier) [reply docs/howto/add_confirm_dialog_in_a_form.md]: "impacting" ?

**Review by bdunogier — APPROVED** [review]

**Pull request merged**